src/app/components/Staking/FinalityProviders/FinalityProviders.tsx

```diff
@@ -49,26 +105,50 @@ export const FinalityProviders: React.FC<FinalityProvidersProps> = ({
         </a>
         .
       </p>
-      <div className="hidden gap-2 px-4 lg:grid lg:grid-cols-stakingFinalityProvidersDesktop">
-        <p>Finality Provider</p>
-        <p>BTC PK</p>
-        <p>Total Delegation</p>
-        <p>Commission</p>
+      <div className="flex gap-3">
+        <FinalityProviderSearch onSearch={handleSearch} />
+        {isMobile && (
+          <button
+            onClick={toggleSortModal}
+            className="btn btn-outline btn-xs text-sm mt-0 font-normal rounded-xl border-gray-400 p-2 h-auto lg:hidden"
+          >
+            <BsSortDown className="text-xl" />
+          </button>
+        )}
       </div>
+      {!isMobile && (
+        <div className="hidden gap-2 px-4 lg:grid lg:grid-cols-stakingFinalityProvidersDesktop">
+          <SortButton
+            field="moniker"
+            label="Finality Provider"
+            onSort={handleSort}
+          />
+          <SortButton field="btcPk" label="BTC PK" onSort={handleSort} />
+          <SortButton
+            field="stakeSat"
+            label="Total Delegation"
+            onSort={handleSort}
+          />
+          <SortButton
+            field="commission"
+            label="Commission"
+            onSort={handleSort}
+          />
+        </div>
+      )}
       <div
         id="finality-providers"
         className="no-scrollbar max-h-[21rem] overflow-y-auto"
       >
         <InfiniteScroll
           className="flex flex-col gap-4"
-          dataLength={finalityProviders?.length || 0}
+          dataLength={sortedProviders.length}
```

Comment:
I would suggest not using `providers` and instead stick with `finalityProviders` explicitly. I.e. `sortedProviders` -> `sortedFinalityProviders`